### ISSUE TITLE
Infinite loop during playback

### ISSUE THREAD

**Issue by Artefact2** [Artefact2/libxm#21]
https://modarchive.org/module.php?33498

Gets stuck in a loop using 100% of the CPU at the 1:18 mark.

~~~
Spd[05/75] Pos[0C/29] Pat[06/32] Row[22/40] Loop[00/01] 00:01:18.88
~~~

**Comment by Artefact2** [comment]
https://modarchive.org/module.php?44847 also gets stuck

~~~
==> Playing: harlot.xm
Spd[03/7D] Pos[08/39] Pat[00/25] Row[1C/40] Loop[00/01] 00:00:32.34
~~~

**Comment by Artefact2** [comment]
~~~
==> Playing: mrgdigi.xm
Spd[06/7D] Pos[0F/22] Pat[12/2C] Row[32/40] Loop[00/01] 00:01:47.91   
~~~

**Comment by Artefact2** [comment]
Fixed by #28, thanks @rasky
